uaclient/security.py

```diff
@@ -1018,15 +1019,25 @@ def _check_subscription_is_expired(cfg: UAConfig) -> bool:
 
     :returns: True if subscription is expired and not renewed.
     """
-    from uaclient import cli
+    effective_to_date = (
+        cfg.machine_token.get("machineTokenInfo", {})
+        .get("contractInfo", {})
+        .get("effectiveTo")
+    )
 
-    try:
-        cli.action_refresh(args=None, cfg=cfg, verbose=False)
-    except exceptions.UserFacingError as e:
-        if status.MESSAGE_ATTACH_EXPIRED_TOKEN in str(e):
-            return not _prompt_for_new_token(cfg)
-        else:
-            raise e
+    if not effective_to_date:
+        msg = (
+            "Cannot verify if the current UA subscription is valid or not\n"
+            "Could not find the effectiveTo field in the UA Contract.\n"
+        )
+        raise exceptions.UserFacingError(msg)
```

Comment:
In what scenario would this happen? Asking because if we know then we might be able to give additional information here.

For example, would it make sense to suggest that the user run `ua refresh` (if that does what I think it does)?

Reply:
That's a good point. 

No, I have never seen this case happen and the code here is just the idea of better be safe than sorry.
But yeah, maybe we should treat this corner case scenario differently. What do you think @blackboxsw ?

Reply:
I think we can drop this check here, I have not seen an absent effectiveTo field in a couple years and the ones we had were temporary test accounts.